pkg/kubelet/prober/prober.go

```diff
@@ -166,7 +166,7 @@ func (pb *prober) runProbe(probeType probeType, p *v1.Probe, pod *v1.Pod, status
 	if p.HTTPGet != nil {
 		scheme := strings.ToLower(string(p.HTTPGet.Scheme))
 		host := p.HTTPGet.Host
-		if host == "" {
+		if host == "" || host == "0.0.0.0" || host == "::" {
```

Comment:
Am I correct that, if we merge this change, than specifying `host = 0.0.0.0` is equivalent to not specifying the host at all?

Thoughts on giving a helpful error message instructing the user to delete the `host` field, instead of automatically falling back to `status.PodIP`, when `0.0.0.0 || ::` is specified?

Especially since specifying `0.0.0.0` is already broken, I don't think =raising an explicit error message would make the the experience any worse for the user. In addition, raising an explicit error message, in my mind, reduces the number of special cases in this code path, which feels easy to reason about.

Just a thought :) Looking forward to discussing! Whatever option we move forward with, it would be great to add unit test coverage so that we can be confident this functionality remains intact going forward.

Reply:
> Am I correct that, if we merge this change, than specifying `host = 0.0.0.0` is equivalent to not specifying the host at all?

yes and no, using 0.0.0.0, at least in Linux, works as destination address
```
curl -v 0.0.0.0:8080
*   Trying 0.0.0.0:8080...
* TCP_NODELAY set
* Connected to 0.0.0.0 (127.0.0.1) port 8080 (#0)
> GET / HTTP/1.1
> Host: 0.0.0.0:8080
> User-Agent: curl/7.65.3
> Accept: */*
>
* Recv failure: Connection reset by peer
* Closing connection 0
curl: (56) Recv failure: Connection reset by peer
```
so is an option that we should "implement the expected behaviour that it has in linux"

> it would be great to add unit test coverage so that we can be confident this functionality remains intact going forward

yeah, I've added the WIP to the title so we know this is not mergeable, I've just wanted to show the changes needed to implement it.